packages/@aws-cdk/aws-events-targets/lib/util.ts

```diff
@@ -45,3 +46,39 @@ export function addLambdaPermission(rule: events.IRule, handler: lambda.IFunctio
     });
   }
 }
+
+
+/**
+ * Allow a rule to send events with failed invocation to an Amazon SQS queue.
+ */
+export function addToDeadLetterQueueResourcePolicy(rule: events.IRule, queue: sqs.IQueue) {
+  const ruleParsedStack = Stack.of(rule);
+  const queueParsedStack = Stack.of(queue);
+
+
+  if (ruleParsedStack.region !== queueParsedStack.region) {
```

Comment:
To compare region and account, you need to use this function:

```ts
/**
 * Whether two string probably contain the same environment dimension (region or account)
 *
 * Used to compare either accounts or regions, and also returns true if both
 * are unresolved (in which case both are expted to be "current region" or "current account").
 */
function sameEnvDimension(dim1: string, dim2: string) {
  return [TokenComparison.SAME, TokenComparison.BOTH_UNRESOLVED].includes(Token.compareStrings(dim1, dim2));
}
```

The values might be tokens and it's not guaranteed they will be the same string in that case.